src/backend/common/helpers/matchstats_helper.py

```diff
@@ -8,27 +8,139 @@
 TTeamIdMap = Dict[TeamId, int]
 
 
-class MatchstatsHelper(object):
+class MatchstatsHelper:
+    OPR_ACCESSOR: StatAccessor = lambda match, color: match.alliances[color]["score"]
+    DPR_ACCESSOR: StatAccessor = lambda match, color: match.alliances[OPPONENT[color]][
+        "score"
+    ]
+    CCWM_ACCESSOR: StatAccessor = lambda match, color: (
+        match.alliances[color]["score"] - match.alliances[OPPONENT[color]]["score"]
+    )
+    DEFAULT_COMPONENT_ACCESSOR: Callable[[Component], StatAccessor] = lambda comp: (
+        lambda match, color: match.score_breakdown[color].get(comp, 0)
+    )
+
+    EVERGREEN: Dict[StatType, StatAccessor] = {
+        StatType.OPR: OPR_ACCESSOR,
+        StatType.DPR: DPR_ACCESSOR,
+        StatType.CCWM: CCWM_ACCESSOR,
+    }
+
+    COMPONENTS = {
+        2020: {
+            "Total Power Cell Points": lambda match, color: (
+                match.score_breakdown[color].get("autoCellPoints")
+                + match.score_breakdown[color].get("teleopCellPoints")
+            ),
+        },
+        2019: {
+            "Cargo + Panel Points": lambda match, color: (
+                match.score_breakdown[color].get("cargoPoints")
+                + match.score_breakdown[color].get("hatchPanelPoints")
+            )
+        }
+    }
+
+    @classmethod
+    def calculate_matchstats(
+        cls, matches: List[Match], skip_coprs: bool = False, keyed: bool = True
+    ) -> EventMatchStats:
+        return EventMatchStats(
+            oprs=cls.calculate_oprs(matches, keyed=keyed),
+            dprs=cls.calculate_dprs(matches, keyed=keyed),
+            ccwms=cls.calculate_ccwms(matches, keyed=keyed),
+            coprs={} if skip_coprs else cls.calculate_coprs(matches, keyed=keyed),
+        )
+
+    @classmethod
+    def calculate_oprs(cls, matches: List[Match], keyed: bool = True) -> TeamStatMap:
+        return cls.calculate_stat(matches, cls.OPR_ACCESSOR, keyed=keyed)
+
+    @classmethod
+    def calculate_dprs(cls, matches: List[Match], keyed: bool = True) -> TeamStatMap:
+        return cls.calculate_stat(matches, cls.DPR_ACCESSOR, keyed=keyed)
+
+    @classmethod
+    def calculate_ccwms(cls, matches: List[Match], keyed: bool = True) -> TeamStatMap:
+        return cls.calculate_stat(matches, cls.CCWM_ACCESSOR, keyed=keyed)
+
+    @classmethod
+    def calculate_coprs(
+        cls, matches: List[Match], keyed: bool = True
+    ) -> Dict[Component, TeamStatMap]:
+        coprs = {}
+
+        # If there is not valid data for COPRs, skip
+        if not (
+            matches is not None
+            and len(matches) > 0
+            and matches[0].score_breakdown is not None
+        ):
+            return coprs
+
+        # for each string-like key in the score_breakdown object
+        # (just take the red score_breakdown from match 0, it's an arbitrary selection)
+        for component, value in matches[0].score_breakdown[AllianceColor.RED].items():
```

Comment:
since you're already checking whether `score_breakdown` is `None` above, this is fine. `pyre` just can't track the nullability of properties (because other side effects can change them).

Do
```
none_throws(matches[0].score_breakdown)[AllianceColor.RED].items()
```